cmake/HPX_AddModule.cmake

```diff
@@ -108,7 +108,7 @@ function(add_hpx_module name)
   target_link_libraries(hpx_${name} ${${name}_DEPENDENCIES})
   target_include_directories(hpx_${name} PUBLIC
     $<BUILD_INTERFACE:${HEADER_ROOT}>
-    $<BUILD_INTERFACE:${CMAKE_BINARY_DIR}>)
+    $<BUILD_INTERFACE:${CMAKE_BINARY_DIR}/include>)
```

Comment:
I think we still need `${CMAKE_BINARY_DIR}` here, *and* `${CMAKE_CURRENT_BINARY_DIR}/include` for the generated headers from this module.

Reply:
Yep !

Reply:
Could you just explain to me why `$<BUILD_INTERFACE:${CMAKE_BINARY_DIR}>` is still necessary ?

Reply:
For the record, we concluded that it isn't necessary.